drivers/adc/ad9656/ad9656.c

```diff
@@ -0,0 +1,217 @@
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include "platform_drivers.h"
+#include "ad9656.h"
+
+/***************************************************************************//**
+ * @brief ad9656_spi_read
+ *******************************************************************************/
+int32_t ad9656_spi_read(struct ad9656_dev *dev,
+			uint16_t reg_addr,
+			uint8_t *reg_data)
+{
+	uint8_t buf[3];
+
+	int32_t ret;
+
+	// the MSB of byte 0 indicates a r/w operation, following 7 bits are the 
+	// bits 14-8 of the address of the register that is accessed. Byte 1 
+	// contains the bits 7-0 of the address of the register.
+	buf[0] = 0x80 | (reg_addr >> 8);
+	buf[1] = reg_addr & 0xFF;
+	buf[2] = 0x00;
+
+	ret = spi_write_and_read(dev->spi_desc,
+				 buf,
+				 3);
+	*reg_data = buf[2];
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief ad9656_spi_write
+ *******************************************************************************/
+int32_t ad9656_spi_write(struct ad9656_dev *dev,
+			 uint16_t reg_addr,
+			 uint8_t reg_data)
+{
+	uint8_t buf[3];
+
+	int32_t ret;
+
+	// the MSB of byte 0 indicates a r/w operation, following 7 bits are the 
+	// bits 14-8 of the address of the register that is accessed. Byte 1 
+	// contains the bits 7-0 of the address of the register. Byte 2 contains
+	// the data to be written.
+	buf[0] = reg_addr >> 8;
+	buf[1] = reg_addr & 0xFF;
+	buf[2] = reg_data;
+
+	ret = spi_write_and_read(dev->spi_desc,
+				 buf,
+				 3);
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief ad9656_setup
+ *******************************************************************************/
+int32_t ad9656_test(struct ad9656_dev *dev,
+		    uint32_t test_mode)
+{
+	ad9656_spi_write(dev,
+			 AD9656_REG_ADC_TEST_MODE,
+			 test_mode);
+	if (test_mode == AD9656_TEST_OFF)
```

Comment:
this can be restructure to 
```
if (test_mode == AD9656_TEST_OFF)
         tmp = AD9656_FORMAT_2S_COMPLEMENT;
else
         tmp = AD9656_FORMAT_OFFSET_BINARY;

return ad9656_spi_write(dev,  AD9656_REG_OUTPUT_MODE, tmp);

Reply:
Done